### ISSUE TITLE
Extract_cluster from DeepCluster_V2 pretrained model

### ISSUE THREAD

**Issue by DC95** [facebookresearch/vissl#364]
Hello @QuentinDuval and @prigoyal 

I was wondering whether I can use the extract_cluster from configs/config/extract_cluster/swav/visualise_swav_resnet_in1k_8gpu.yaml for **deep_clusterV2**  pretrained model?

I tried with one of the basic changes with the **head configuration** i.e. I tried the one which I used in my pretraining- 

    HEAD:
       PARAMS: [
         ["mlp", {"dims": [2048, 2048], "use_relu": True, "skip_last_layer_relu_bn": False}],
         ["mlp", {"dims": [2048, 128]}],
       ]
      
But it came up with the error - 

> File "/p/project/deepacf/kiste/DC/vissl/vissl/trainer/trainer_main.py", line 514, in extract_clusters
    assert self.task.base_model.is_clustering_model(), error_message
AssertionError: Extracting clusters is only available for pre-training methods based on clusters

**My understanding -** 

I looked at the code from **trainer_main.py** and the function **def extract_clusters(self)**,  here it is mentioned - 

_extract multi-GPU cluster extraction for pre-trained models based on clusterization (SwAV, **DeepCluster**, etc.)_

But I also went to function **def [is_clustering_model**](https://github.com/facebookresearch/vissl/blob/6c21cd91e66dcf74f4533463977514bb11a2cd1e/vissl/models/base_ssl_model.py#L551) (used for asserting the base model line 514 in def extract_clusters) where it is written -

_- only supported for **SWAV currently**_ (This contradicts the above)

I think extract_cluster is not yet ready for deepcluster_v2. Can somehow we do it??

**Comment by QuentinDuval** [comment]
Hi @DC95,

Indeed, the extract cluster is for the moment only working for SwAV and the message is a bit misleading.

I think we can schedule an enhancement on our side to improve this and support DeepClusterV2 in addition to SwAV (@prigoyal what do you think?)?

Or if you feel comfortable trying to contribute it (it seems you went through some of the code already), give it a go and open a PR with it: we will very much welcome your help :)

**Comment by DC95** [comment]
Dear @QuentinDuval and @prigoyal 

I have not done this before :) but I will give it a go. I will definitely need your support. 

Regards,
DC

**Comment by DC95** [comment]
Dear @QuentinDuval 

From today, I have started to look deeper into the codes and management of VISSL. But I see that you have self-assigned it yourself. 

Of course, you are the best person to do this job. But If I can help you in any way, I will be glad to.

In my Ph.D. research, I am using the DC_v2 model for assessing satellite images. Therefore I am keenly interested in this development.

Regards,
DC

**Comment by QuentinDuval** [comment]
> Dear @QuentinDuval
> 
> From today, I have started to look deeper into the codes and management of VISSL. But I see that you have self-assigned it yourself.
> 
> Of course, you are the best person to do this job. But If I can help you in any way, I will be glad to.
> 
> In my Ph.D. research, I am using the DC_v2 model for assessing satellite images. Therefore I am keenly interested in this development.
> 
> Regards,
> DC

Hi @DC95,

Of course, please go ahead :) I assigned it to you. We would be super glad to get your contribution!

I very am sorry for the confusion (what I did is assigned it to myself temporarily, mostly because we wanted to sort out the issues and get rid of those that are "unassigned"). Sorry for that !

Please go ahead with your proposal, I would be glad to review your PR :)

**Comment by DC95** [comment]
Dear @QuentinDuval 

I managed to submit a [PR](https://github.com/facebookresearch/vissl/pull/417).

I think there was one main shortcoming in this process - 

1. To take care of multi heads for the multi clustering in DC_V2

I don't know how much I have succeeded in this pr. I can definitely say It's not correct as in my case; I am giving multi clustering of num_clusters - [10,10,10]

But then, in the final results, the no of clusters is extending 10 (up to some random 108 clusters). 

Regards,
DC
